packages/bloc_lint/lib/src/rules/avoid_context_extensions.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:bloc_lint/bloc_lint.dart';
+
+/// {@template avoid_context_extensions}
+/// The avoid_context_extensions lint rule.
+/// {@endtemplate}
+class AvoidContextExtensions extends LintRule {
+  /// {@macro avoid_context_extensions}
+  AvoidContextExtensions([Severity? severity])
+    : super(name: rule, severity: severity ?? Severity.warning);
+
+  /// The name of the lint rule.
+  static const rule = 'avoid_context_extensions';
+
+  @override
+  Listener? create(LintContext context) {
+    return _Listener(context);
+  }
+}
+
+class _Listener extends Listener {
+  _Listener(this.context);
+
+  static const _contextExtensions = [
```

Comment:
I don’t think this is quite right because someone can be using pkg:provider and should still be able to use context extensions on non-bloc objects without any lints

Reply:
Yeap, you are right. I've just updated the implementation to address your point, and it now only flags context extensions on types whose names end with the `Bloc` or `Cubit` suffix, which avoids incorrect lints on other packages like provider. I had to rely on this naming convention because, with the current parser listener architecture, it isn't feasible to inspect the full class hierarchy to programmatically verify if a type extends `Bloc` or extends `Cubit`. Let me know what you think